sphinx/texinputs/sphinxlatexoptionhandling.sty

```diff
@@ -128,7 +128,12 @@
 \DisableKeyvalOption{sphinx}{numfigreset}
 \DisableKeyvalOption{sphinx}{nonumfigreset}
 \DisableKeyvalOption{sphinx}{mathnumfig}
+% To allow hyphenation of first word in narrow contexts; no option,
+% customization to be done via 'preamble' key
+\newcommand*\sphinxAtStartPar{\nobreak\hskip\z@skip}
+% No need for the \hspace{0pt} trick (\hskip\z@skip) with luatex
+\ifdefined\directlua\let\sphinxAtStartPar\@empty\fi
```

Comment:
@jfbu This hunk is copied manually. Could you check the recent updates are surely merged into master (because we need to separate the changes on 3.x on merging into the master)?